Gems/Atom/Feature/Common/Code/Source/CubeMapCapture/CubeMapRenderer.cpp

```diff
@@ -0,0 +1,116 @@
+
+#include <CubeMapCapture/CubeMapRenderer.h>
+#include <Atom/RPI.Public/Pass/Pass.h>
+#include <Atom/RPI.Public/Pass/PassSystemInterface.h>
+#include <Atom/RPI.Public/RenderPipeline.h>
+#include <Atom/RPI.Public/View.h>
+#include <Atom/RPI.Reflect/Pass/EnvironmentCubeMapPassData.h>
+
+namespace AZ
+{
+    namespace Render
+    {
+        void CubeMapRenderer::StartRender(RenderCubeMapCallback callback, const AZ::Transform& transform, float exposure)
+        {
+            AZ_Assert(m_scene, "CubeMapRenderer::StartRender called without a valid scene");
+            AZ_Assert(m_renderingCubeMap == false, "CubeMapRenderer::StartRender called while a cubemap render was already in progress");
+            if (m_renderingCubeMap)
+            {
+                return;
+            }
+
+            m_renderingCubeMap = true;
+            m_callback = callback;
+            m_exposure = exposure;
+
+            AZ::RPI::RenderPipelineDescriptor environmentCubeMapPipelineDesc;
+            environmentCubeMapPipelineDesc.m_mainViewTagName = "MainCamera";
+            environmentCubeMapPipelineDesc.m_renderSettings.m_multisampleState = RPI::RPISystemInterface::Get()->GetApplicationMultisampleState();
+            environmentCubeMapPipelineDesc.m_renderSettings.m_size.m_width = RPI::EnvironmentCubeMapPass::CubeMapFaceSize;
+            environmentCubeMapPipelineDesc.m_renderSettings.m_size.m_height = RPI::EnvironmentCubeMapPass::CubeMapFaceSize;
+
+            // create a unique name for the pipeline
+            AZ::Uuid uuid = AZ::Uuid::CreateRandom();
+            AZStd::string uuidString;
+            uuid.ToString(uuidString);
+            environmentCubeMapPipelineDesc.m_name = AZStd::string::format("EnvironmentCubeMapPipeline_%s", uuidString.c_str());
+            RPI::RenderPipelinePtr environmentCubeMapPipeline = AZ::RPI::RenderPipeline::CreateRenderPipeline(environmentCubeMapPipelineDesc);
+            m_environmentCubeMapPipelineId = environmentCubeMapPipeline->GetId();
+
+            AZStd::shared_ptr<RPI::EnvironmentCubeMapPassData> passData = AZStd::make_shared<RPI::EnvironmentCubeMapPassData>();
+            passData->m_position = transform.GetTranslation();
+
+            RPI::PassDescriptor environmentCubeMapPassDescriptor(Name("EnvironmentCubeMapPass"));
+            environmentCubeMapPassDescriptor.m_passData = passData;
+
+            m_environmentCubeMapPass = RPI::EnvironmentCubeMapPass::Create(environmentCubeMapPassDescriptor);
+            m_environmentCubeMapPass->SetRenderPipeline(environmentCubeMapPipeline.get());
+
+            const RPI::Ptr<RPI::ParentPass>& rootPass = environmentCubeMapPipeline->GetRootPass();
+            rootPass->AddChild(m_environmentCubeMapPass);
+
+            // store the current exposure values
+            Data::Instance<RPI::ShaderResourceGroup> sceneSrg = m_scene->GetShaderResourceGroup();
+            m_previousGlobalIblExposure = sceneSrg->GetConstant<float>(m_globalIblExposureConstantIndex);
```

Comment:
Question: Are we possibly reading write-combined memory here (possibly aperture memory in-flight on the GPU?). I feel like I would be a bit more comfortable if we shadowed updates to this data in strictly CPU visible memory instead of reading it from the SRG directly, which would make this code robust if the SRG memory type were to change now or in the future.

Reply:
Nvm, it looks like all SRG constant data already maintains a CPU-visible non-wc copy.

Reply:
I'm using the RPI SRG public interface here so I would think this is supported behavior.  In the end it's reading it back out of the SRG data buffer, which is the CPU side buffer.